### PR TITLE
raft: raft learners should be returned after applyConfChange

### PR DESCRIPTION
**Pull request by july2993**
fix https://github.com/pingcap/raft-rs/issues/12

add a new test test_raw_node_propose_add_learner_node() in test_raw_node.rs to cover the fix in apply_conf_change(..)

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: tests/cases/test_raw_node.rs [hunk 1]
```diff
@@ -269,6 +269,42 @@ fn test_raw_node_propose_add_duplicate_node() {
...
+    }
+
+    // propose add learner node and check apply state
+    let cc = conf_change(ConfChangeType::AddLearnerNode, 2);
+    raw_node.propose_conf_change(cc).expect("");
+
+    let rd = raw_node.ready();
+    s.wl().append(&rd.entries).expect("");
+    for e in rd.committed_entries.as_ref().unwrap() {
+        if e.get_entry_type() == EntryType::EntryConfChange {
```
Comment (BusyJay) [inline tests/cases/test_raw_node.rs]: What if conf change entry is not found after iterating all entries?

Comment (july2993) [reply tests/cases/test_raw_node.rs]: updated: check the committed_entries explicitly

**Comment by july2993** [comment]
@BusyJay test_remove_learner is updated now

**Comment by BusyJay** [comment]
Thanks!

**Pull request merged**